FakeXrmEasy.Tests.Shared/FakeContextTests/RemoveFromQueueRequestTests/RemoveFromQueueRequestTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿#if FAKE_XRM_EASY_2013 || FAKE_XRM_EASY_2015 || FAKE_XRM_EASY_2016 || FAKE_XRM_EASY_365 || FAKE_XRM_EASY_9
+
+using FakeXrmEasy.FakeMessageExecutors;
+using Microsoft.Crm.Sdk.Messages;
+using Microsoft.Xrm.Sdk;
+using Microsoft.Xrm.Sdk.Messages;
+using System;
+using System.Linq;
+using System.ServiceModel;
+using Xunit;
+
+namespace FakeXrmEasy.Tests.FakeContextTests.RemoveFromQueueRequestTests
+{
+    public class RemoveFromQueueRequestTests
+    {
+        [Fact]
+        public void When_can_execute_is_called_with_an_invalid_request_result_is_false()
+        {
+            var executor = new RemoveFromQueueRequestExecutor();
+            var anotherRequest = new RetrieveMultipleRequest();
+            Assert.False(executor.CanExecute(anotherRequest));
+        }
+
+        [Fact]
+        public void When_a_request_is_called_Queueitem_Is_Removed()
+        {
+            var context = new XrmFakedContext();
+
+            var queueItem = new Entity
+            {
+                LogicalName = Crm.QueueItem.EntityLogicalName,
+                Id = Guid.NewGuid()
+            };
+
+            context.Initialize(new[]
+            {
+                queueItem
+            });
+
+            var executor = new RemoveFromQueueRequestExecutor();
+
+            var req = new RemoveFromQueueRequest
+            {
+                QueueItemId = queueItem.Id
+            };
+
+            executor.Execute(req, context);
+
+            var retrievedQueueItem = context.Data[Crm.QueueItem.EntityLogicalName].Values;
```

Comment:
@Kraviecc , another option for querying data is using the XrmFakedContext.CreateQuery("entityname").ToList() or any other LINQ expression. This is recommended instead of accessing the context.Data directly, just in case the context.Data implementation changes internally, following separation of concerns patterns.

Reply:
thanks for pointing that :) I've seen this way somewhere in code and decided to try it.

Reply:
You're welcome :)